src/third_party/README.md

```diff
@@ -0,0 +1,49 @@
+# Third Party Libraries
+
+Genny depends on a set of third party libraries to compile and run.
+These libraries are included in one of three general ways:
+1. Toolchain libraries via find_package
+2. Vendored libraries via add_subdirectory
+3. User libraries via find_package
+
+## Toolchain Libaries
+
+Toolchain libraries are generally monolithic pieces of code with a large collection of headers.
+If they can be linked via a static library, they should be. Generally speaking, these libraries are
+expected to have a version that is acceptable for genny for the foreseeable future. Often, these
+libraries are so essential that the CMake find_package module searches for their core headers at
+known locations. For alternative locations, usually the module will allow the user to specify a
+prefix dir. While distributing a specific version of these libraries may be necessary, there will
+almost always be a known recipe for building them for a given distro.
+
+Genny currently uses:
+* libstdc++ (with c++ 17)
+* libicuuc
+* Boost (with log headers)
+* OpenSSL 1.1
+
+## Vendored Libraries
+
+Vendored libraries are usually small libraries that facilitate a unique capability. 
+
+Libraries should be vendored in when they fit the following criteria:
+* They have no additional dependencies beyond what genny needs via its system libraries.
+* They are expected to change version rarely. 
+* They can be statically linked or compiled in as object files.
+
+Genny currently uses:
+* yaml-cpp
+* Loki
+* Catch2
+* Jasper
+
+## User Libraries
+
+User libraries are modular pieces with a known ABI and some dependencies. Usually, linking to
```

Comment:
So I understand from this comment how user libraries are functionally different to toolchain libraries; does that mean we dynamically link them due to convenience? Asking because I'm about to change the genny toolchain to be all vcpkg and I'd like to include the user libraries in there as well.

Reply:
That's a good question. So there are two concerns here, one that we care about quite a bit, and the other one that we surprisingly don't care about very much:
* User libraries usually have extra deps to other libraries. (We've deliberately limited that with our distributable builds of these libraries.) We shared link because we may or may not have those dependency libraries linked into our executable by default and we'd rather not care about these deps even if the cmake module pulls them in. For mongocxx, that's mongoc. For grpc, that's protobuf and c-ares.
* People get touchy around static linking and licenses. (See [here](https://softwareengineering.stackexchange.com/questions/167773/how-does-the-gpl-static-vs-dynamic-linking-rule-apply-to-interpreted-languages) Another hallmark of libraries that fit this bill is that they may have some restrictive licensing conditions. Somehow, ours are both Apache. It's weird.

Reply:
I think I should put in a section on linking in general here.

Reply:
This is good and well-reasoned info - would love a touch of color on linking since I don't have a good intuition about what to link shared vs static and I suspect it's not actually that complicated in the general case.